AX: Voiceover/Safari is not reading the contents of divs with role=“tooltip”

twilco opened the pull request:
#### 70128b4a24b6ff0cebc5c231354e78337735981d
<pre>
AX: Voiceover/Safari is not reading the contents of divs with role=“tooltip”
<a href="https://bugs.webkit.org/show_bug.cgi?id=276690">https://bugs.webkit.org/show_bug.cgi?id=276690</a>
<a href="https://rdar.apple.com/131877635">rdar://131877635</a>

Reviewed by Chris Fleizach.

When the text underneath the target of an aria-describedby changes, we need to post AXExtendedDescriptionChanged to
ensure the accessibility tree is updated.

* LayoutTests/accessibility/dynamic-aria-describedby-text-expected.txt: Added.
* LayoutTests/accessibility/dynamic-aria-describedby-text.html: Added.
* LayoutTests/platform/ios/TestExpectations: Enable new test.
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::handleChildrenChanged):
(WebCore::AXObjectCache::updateIsolatedTree):

Canonical link: <a href="https://commits.webkit.org/282430@main">https://commits.webkit.org/282430@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9b9a41103d7c9f9f3246ea80882ba0ea31aeaac1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63120 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42476 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15716 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67141 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13728 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65240 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50163 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14008 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50859 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9471 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66189 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39444 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54649 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31545 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36129 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11996 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12600 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57669 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12325 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/68836 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7066 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/11939 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58174 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7098 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54721 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58392 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13991 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5889 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38296 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39376 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40487 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39118 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fleizach commented on LayoutTests/accessibility/dynamic-aria-describedby-text.html:
```diff
@@ -0,0 +1,43 @@
+        document.getElementById("description").innerHTML = `
+            <div>Final text</div>
+        `;
+        output += await expectAsync("button.customContent.includes('Final text')", "true");
```
do we need to test the sending of the notification?

twilco replied:
Thanks for reviewing! I don't think so — the notification is internal only, only used as a signal to update the accessibility tree.

twilco committed:
AX: Voiceover/Safari is not reading the contents of divs with role=“tooltip”
https://bugs.webkit.org/show_bug.cgi?id=276690
rdar://131877635

Reviewed by Chris Fleizach.

When the text underneath the target of an aria-describedby changes, we need to post AXExtendedDescriptionChanged to
ensure the accessibility tree is updated.

* LayoutTests/accessibility/dynamic-aria-describedby-text-expected.txt: Added.
* LayoutTests/accessibility/dynamic-aria-describedby-text.html: Added.
* LayoutTests/platform/ios/TestExpectations: Enable new test.
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::handleChildrenChanged):
(WebCore::AXObjectCache::updateIsolatedTree):

Canonical link: https://commits.webkit.org/282430@main
